Source/WebSocket_Windows.cpp

```diff
@@ -0,0 +1,135 @@
+#include "WebSocket_Base.h"
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <Windows.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <winrt/Windows.Networking.Sockets.h>
+#include <arcana/threading/task_conversions.h>
+
+namespace UrlLib
+{
+    using namespace winrt;
+
+    class WebSocket::WSImpl : public WSImplBase
```

Comment:
>class WebSocket::WSImpl : public WSImplBase [](http://example.com/codeflow?start=4&length=43)

is there a reason why these class decls are not in their own headers? (is this some convention in UrlLib?)

Reply:
Yeah it's because of the include order. This follows the same format as UrlRequest (babylon's existing code).

Reply:
Yes, this is a form of the [pimpl](https://en.cppreference.com/w/cpp/language/pimpl) pattern.